ads/views.py

```diff
@@ -39,6 +41,30 @@ class AdDetailView(DetailView):
     def get_object(self):
         return get_object_or_404(Ads, slug=self.kwargs['ad_slug'], category__slug=self.kwargs['category_slug'])
     
+    def post(self, request, *args, **kwargs):
+        self.ad = self.get_object()
+
+        if self.ad.user.id > self.request.user.id:
+            user1, user2 = self.request.user, self.ad.user
+        else:
+            user1, user2 = self.ad.user, self.request.user
+
+        conversation_id = f"{user1.id}-{user1.username}-{self.ad.slug}-{user2.id}-{user2.username}"
+
+        user_conversations = get_user_conversations(request.user)
+        conversation_ids = [conv['conversation_id'] for conv in user_conversations]
```

Comment:
Not able to understand. Make it readable, extract it to another meaningful method. One should be able to understand what is happening here.

Reply:
ok sir